lib/connect.ts

```diff
@@ -65,1354 +69,648 @@ import utils from "./utils";
         }
 
-        if (data.data instanceof Array) {
-            data.data = results;
+        // Set request header content type
+        if (isJSON) {
+            options['headers']['Content-Type'] = 'application/json';
         } else {
-            data.data = results[0];
+            options['headers']['Content-Type'] = 'application/x-www-form-urlencoded';
+        }
+        // Set response transformer
+        if (responseTransformer) {
+            options.transformResponse = (axios.defaults.transformResponse as any).concat(responseTransformer);
         }
 
-        return data;
+        return this.requestInstance.request(options);
     }
 
-    function parseHistorical(jsonData) {
-        // Return if its an error
-        if (jsonData.error_type) return jsonData;
-
-        const results = [] as any[];
-        for(let i=0; i<jsonData.data.candles.length; i++) {
-            const d = jsonData.data.candles[i];
-            const c = {
-                "date": new Date(d[0]),
-                "open": d[1],
-                "high": d[2],
-                "low": d[3],
-                "close": d[4],
-                "volume": d[5]
-            }
 
-            // Add OI field if its returned
-            if (d[6]) {
-                c["oi"] = d[6]
-            }
 
-            results.push(c);
-        }
 
-        return { "data": results };
+
+}
+
+function formatGenerateSession(data: AnyObject): AnyObject {
+    if (!data.data || typeof data.data !== 'object') return data;
+
+    if (data.data.login_time) {
+        data.data.login_time = new Date(data.data.login_time);
     }
 
-    function transformInstrumentsResponse(data, headers) {
-        // Parse CSV responses
-        if (headers["content-type"] === "text/csv") {
-            const parsedData = csvParse.parse(data, {"header": true}).data;
-            for (const item of parsedData as any) {
-                item["last_price"] = parseFloat(item["last_price"]);
-                item["strike"] = parseFloat(item["strike"]);
-                item["tick_size"] = parseFloat(item["tick_size"]);
-                item["lot_size"] = parseInt(item["lot_size"]);
-
-                if (item["expiry"] && item["expiry"].length === 10) {
-                    item["expiry"] = new Date(item["expiry"]);
-                }
-            }
+    return data;
+}
+
+function formatQuoteResponse(data: AnyObject): AnyObject {
+    if (!data.data || typeof data.data !== 'object') return data;
 
-            return parsedData;
+    for (const k in data.data) {
+        const item = data.data[k];
+        for (const field of ['timestamp', 'last_trade_time']) {
+            if (item[field] && item[field].length === 19) {
+                item[field] = new Date(item[field]);
+            }
         }
+    }
 
-        return data;
+    return data;
+}
+
+// Format response ex. datetime string to date
+function formatResponse(data: AnyObject): AnyObject {
```

Comment:
It is advisable to designate the `formatResponse` function as a private function.